OneDriveApiBrowser/FormBrowser.cs

```diff
@@ -271,71 +281,87 @@ private void UpdateConnectedStateUx(bool connected)
 
         private async Task SignIn(ClientType clientType)
         {
-            if (this.oneDriveClient == null)
+            if (clientType == ClientType.Consumer)
```

Comment:
It's a little funky because the `MsaAuthenticationProvider` and the `AdalAuthenticationProvider` both implement `IAuthenticationProvider` from Microsoft.Graph, but that doesn't include `AuthenticateUserAsync()`, so they have no common ancestor, plus they have different call patterns (only the ADAL version requires a target URL). They are different enough that it seems to make sense to separate them. :/ Would love to discuss more if you have insight I missed.